uaclient/apt.py

```diff
@@ -621,3 +645,35 @@ def get_apt_cache_datetime() -> Optional[datetime.datetime]:
     if cache_time is None:
         return None
     return datetime.datetime.fromtimestamp(cache_time, datetime.timezone.utc)
+
+
+def update_esm_caches() -> None:
+    if not system.is_current_series_lts():
+        return
+
+    from apt import Cache  # type: ignore
+
+    from uaclient.entitlements.entitlement_status import ApplicationStatus
+    from uaclient.entitlements.esm import (
+        ESMAppsEntitlement,
+        ESMInfraEntitlement,
+    )
+
+    # Always setup ESM-Apps
+    if (
+        ESMAppsEntitlement().application_status()[0]
```

Comment:
All of those calls will create new `UAConfig` objects. Should we make the `update_esm_cache` receive an instance of that object ?